src/components/DimensionsPanel/List/__tests__/DimensionItem.spec.js

```diff
@@ -1,51 +1,45 @@
-import { shallow } from 'enzyme'
+import { render } from '@testing-library/react'
 import React from 'react'
 import DimensionItem from '../DimensionItem.js'
 
-describe('DimensionItem', () => {
-    let props
-    let shallowItem
-    const dimensionItem = () => {
-        if (!shallowItem) {
-            shallowItem = shallow(<DimensionItem {...props} />)
-        }
-        return shallowItem
-    }
-
-    beforeEach(() => {
-        props = {
-            id: 'pe',
-            name: 'Period',
-            isDeactivated: false,
-            isSelected: false,
-            isRecommended: false,
-            isLocked: false,
-        }
-
-        shallowItem = undefined
-    })
-
-    it('matches the snapshot', () => {
-        expect(dimensionItem()).toMatchSnapshot()
-    })
-
-    it('matches the snapshot with recommended', () => {
-        props.isRecommended = true
-        expect(dimensionItem()).toMatchSnapshot()
-    })
-
-    it('matches the snapshot with selected', () => {
-        props.isSelected = true
-        expect(dimensionItem()).toMatchSnapshot()
-    })
-
-    it('matches the snapshot with locked', () => {
-        props.isLocked = true
-        expect(dimensionItem()).toMatchSnapshot()
-    })
-
-    it('matches the snapshot with onOptionsClick', () => {
-        props.onOptionsClick = jest.fn()
-        expect(dimensionItem()).toMatchSnapshot()
-    })
+const props = {
```

Comment:
I think it would be cleaner to have a distinct props const in each test. As it is now, each test affects the next (i.e., isRecommended is set to true in the second test, and remains true for the following tests)

Reply:
Fixed.
I kept the "shared" props object but instead of changing values there I pass the specific options directly in the component.